File: app/src/main/java/com/easy/view/canvas/geometry/view/CustomTriangleView.java
package com.easy.view.canvas.geometry.view;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.ColorInt;
import androidx.annotation.Nullable;
import androidx.core.graphics.PaintKt;

/**
 * Created by shishoufeng on 2020/9/6.
 * email:shoufengshi@lagou.com
 * <p>
 * desc: 自定义 三角形view
 */
public class CustomTriangleView extends View {

    private Context mContext;

    private Paint paint;

    @ColorInt
    private int paintColor = Color.BLACK;

    public CustomTriangleView(Context context) {
        this(context, null);
    }

    public CustomTriangleView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public CustomTriangleView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        this.mContext = context;


        paint = new Paint();

    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);


        paint.setColor(paintColor);






    }

    public int getPaintColor() {
        return paintColor;
    }

    public void setPaintColor(int paintColor) {
        this.paintColor = paintColor;
    }
}

File: app/src/main/java/com/easy/view/canvas/geometry/view/CustomTriangleView.kt
package com.easy.view.canvas.geometry.view

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View
import androidx.annotation.ColorInt

/**
 * Created by shishoufeng on 2020/9/6.
 * email:shoufengshi@lagou.com
 *
 *
 * desc: 自定义 三角形view
 */
class CustomTriangleView @JvmOverloads constructor(
    private val mContext: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : View(
    mContext, attrs, defStyleAttr
) {
    private val paint: Paint

    @ColorInt
    var paintColor = Color.BLACK
    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        paint.color = paintColor
    }

    init {
        paint = Paint()
    }
}